tools/stub_generator/stub_codegen.py

```diff
@@ -89,28 +95,29 @@ def main():
         if cursor.kind != clang.cindex.CursorKind.FUNCTION_DECL:
             continue
 
+        function_name = cursor.spelling
+
         # make sure that we deal  only with functions with no definition
-        if cursor.spelling not in config['functions'] or cursor.spelling in all_definition or \
-           cursor.spelling not in all_declaration:
+        if function_name not in config['functions'] or function_name in all_definition or \
+           function_name not in all_declaration:
             continue
 
         # make sure that we deal with every function only once
-        all_declaration.remove(cursor.spelling)
-
-        with open(cursor.location.file.name, 'r', encoding='latin-1') as file:
-            start = cursor.extent.start.offset
-            end = cursor.extent.end.offset
-            declaration = file.read()[start:end]
```

Comment:
This copied the declaration from the original file - it didn't work correctly, because there might have been some macros - and they could be used in different order, yielding different results:
```C++
#define cuFileDriverClose cuFileDriverClose_v2
CUFileError_t cuFileDriverClose();
```
or
```C++
CUFileError_t cuFileDriverClose_v2();
#define cuFileDriverClose cuFileDriverClose_v2
```